packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_TentativePFactory_def.hpp

```diff
@@ -520,7 +530,8 @@ namespace MueLu {
 
             for (size_t k = 0, lnnz = 0; k < NSDim; k++) {
               // Skip zeros (there may be plenty of them, i.e., NSDim > 1 or boundary conditions)
-              const SC qr_jk = fineNS[k][rowMap->getLocalElement(aggToRowMapGO[aggStart[agg]+j])];
+              SC qr_jk = fineNS[k][rowMap->getLocalElement(aggToRowMapGO[aggStart[agg]+j])];
+              if(constantColSums) qr_jk = qr_jk / (double)aggSizes[agg];
```

Comment:
Having all these if's is a little bit annoying from a performance perspective. Is it easy to duplicate loops ... which of course isn't great from a maintenance perspective.

Reply:
One would hope that branch prediction --- of a const --- would do the right thing here.

Reply:
Yeah ... I agree.